oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticCustomAnalyzerMappings.java

```diff
@@ -112,78 +112,30 @@ protected interface ParameterTransformer {
         LUCENE_ELASTIC_TRANSFORMERS = new LinkedHashMap<>();
 
         LUCENE_ELASTIC_TRANSFORMERS.put(WordDelimiterFilterFactory.class, luceneParams -> {
-            if (luceneParams.containsKey("generateWordParts")) {
-                luceneParams.put("generateWordParts", Integer.parseInt(luceneParams.get("generateWordParts").toString()) == 1);
-            }
-            if (luceneParams.containsKey("generateNumberParts")) {
-                luceneParams.put("generateNumberParts", Integer.parseInt(luceneParams.get("generateNumberParts").toString()) == 1);
-            }
-            if (luceneParams.containsKey("catenateWords")) {
-                luceneParams.put("catenateWords", Integer.parseInt(luceneParams.get("catenateWords").toString()) == 1);
-            }
-            if (luceneParams.containsKey("catenateNumbers")) {
-                luceneParams.put("catenateNumbers", Integer.parseInt(luceneParams.get("catenateNumbers").toString()) == 1);
-            }
-            if (luceneParams.containsKey("catenateAll")) {
-                luceneParams.put("catenateAll", Integer.parseInt(luceneParams.get("catenateAll").toString()) == 1);
-            }
-            if (luceneParams.containsKey("splitOnCaseChange")) {
-                luceneParams.put("splitOnCaseChange", Integer.parseInt(luceneParams.get("splitOnCaseChange").toString()) == 1);
-            }
-            if (luceneParams.containsKey("preserveOriginal")) {
-                luceneParams.put("preserveOriginal", Integer.parseInt(luceneParams.get("preserveOriginal").toString()) == 1);
-            }
-            if (luceneParams.containsKey("splitOnNumerics")) {
-                luceneParams.put("splitOnNumerics", Integer.parseInt(luceneParams.get("splitOnNumerics").toString()) == 1);
-            }
-            if (luceneParams.containsKey("stemEnglishPossessive")) {
-                luceneParams.put("stemEnglishPossessive", Integer.parseInt(luceneParams.get("stemEnglishPossessive").toString()) == 1);
-            }
+            Consumer<String> transformFlag = flag -> {
+                if (luceneParams.containsKey(flag)) {
+                    luceneParams.put(flag, Integer.parseInt(luceneParams.get(flag).toString()) == 1);
+                }
```

Comment:
I think this can be done with 
```
luceneParms.computeIfPresent(flag, Integer.parseInt(luceneParams.get(flag).toString() == 1)
```
https://docs.oracle.com/en%2Fjava%2Fjavase%2F11%2Fdocs%2Fapi%2F%2F/java.base/java/util/Map.html#computeIfPresent(K,java.util.function.BiFunction)